File: app/src/main/java/com/feriantes4dawin/feriavirtualmovil/data/model/Usuario.java
package com.feriantes4dawin.feriavirtualmovil.data.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.feriantes4dawin.feriavirtualmovil.ui.util.FeriaVirtualConstants;


@Entity(tableName= "usuario")
public class Usuario {
    @PrimaryKey(autoGenerate = false)
    public Integer secret_id_usuario = FeriaVirtualConstants.CURRENT_LOGGED_USUARIO;
    public Integer id_usuario;
    public String personal_id;
    public String nombre;
    public String nombre_segundo;
    public String apellido_materno;
    public String apellido_paterno;
    public String direccion;
    public Integer telefono;
    public String email;
    public String contrasena;
    public String estado_contrato;
    public String estado_usuario;
    public String nacionalidad;
    public String rol;

}

File: app/src/main/java/com/feriantes4dawin/feriavirtualmovil/data/model/Usuario.kt
package com.feriantes4dawin.feriavirtualmovil.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

/*Esto permite tener un solo usuario en la base de datos
* y así asegurar que el que tiene iniciada la sesión sea
* único.
*/
const val CURRENT_LOGGED_USUARIO :Int = 0

@Entity(tableName= "usuario")
data class Usuario(
    @PrimaryKey(autoGenerate = false)
    val secret_id_usuario: Int = CURRENT_LOGGED_USUARIO,
    val id_usuario: Int,
    val personal_id: String,
    val nombre: String,
    val nombre_segundo: String,
    val apellido_materno: String,
    val apellido_paterno: String,
    val direccion: String,
    val telefono: Int,
    val email: String,
    val contrasena:String,
    val estado_contrato: String,
    val estado_usuario: String,
    val nacionalidad: String,
    val rol: String
)